cloud-controller-manager/do/doks_lb_validator.go

```diff
@@ -0,0 +1,206 @@
+	if svc.Spec.Type == v1.ServiceTypeLoadBalancer && svc.DeletionTimestamp != nil {
+		return admission.Allowed("the loadbalancer is deleting")
+	}
+
+	// initialize DO Client if not already initialized
+	if v.gClient == nil {
+		doClient, err := initDOClient()
+		if err != nil {
+			v.Log.Error(err, "failed to initialize DO client")
+			return admission.Errored(http.StatusConflict, err)
+		}
+		v.gClient = doClient
+	}
+
+	clusterID := os.Getenv(doClusterIDEnv)
+	if clusterID == "" {
+		fmt.Println("missing cluster id")
+	}
+	region := "nyc1"
+	if region == "" {
+		fmt.Println("missing region id")
+	}
+
+	nodePools, _, err := v.gClient.Kubernetes.ListNodePools(ctx, clusterID, &godo.ListOptions{})
+	if err != nil {
+		v.Log.Error(err, "no nodes found")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	var dropletIDs []int
+	for _, pool := range nodePools {
+		for _, node := range pool.Nodes {
+			dropletID, err := strconv.Atoi(node.DropletID)
+			if err != nil {
+				v.Log.Error(err, "failed to retrieve droplet ids")
+				return admission.Errored(http.StatusConflict, err)
+			}
+			dropletIDs = append(dropletIDs, dropletID)
+		}
+	}
+
+	forwardingRules := []godo.ForwardingRule{
+		{
+			EntryProtocol:  "http",
+			EntryPort:      80,
+			TargetProtocol: "http",
+			TargetPort:     80,
+			CertificateID:  "",
+			TlsPassthrough: false,
+		}}
+
+	lbRequest, err := v.buildRequest(svc.Name, region, dropletIDs, forwardingRules)
+	if err != nil {
+		v.Log.Error(err, "failed to build lb request")
+		//fmt.Printf("failed to build lb request", err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	err = v.decoder.DecodeRaw(req.OldObject, svc)
+	if err != nil {
+		err = v.validateCreate(ctx, lbRequest, v.gClient)
+		if err != nil {
+			v.Log.Error(err, "failed to validate lb creation")
+			return admission.Denied("failed to validate load balancer request")
+		}
+		return admission.Allowed("valid lb create request")
+	}
+
+	err = v.validateUpdate(ctx, svc, lbRequest, v.gClient)
+	if err != nil {
+		fmt.Printf("failed to update load balancer")
+		v.Log.Error(err, "failed to update load balancer")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	return admission.Allowed("validating if request type is UPDATE")
+}
+
+func (v *DOKSLBServiceValidator) validateCreate(ctx context.Context, lbRequest *godo.LoadBalancerRequest, doClient *godo.Client) error {
+	_, _, err := doClient.LoadBalancers.Create(ctx, lbRequest)
+	if err != nil {
+		v.Log.Error(err, "failed to create load balancer")
+		return err
+	}
+
+	return nil
+}
+
+func (v *DOKSLBServiceValidator) validateUpdate(ctx context.Context, svc *v1.Service, lbRequest *godo.LoadBalancerRequest, doClient *godo.Client) error {
+	currentLBID := svc.Annotations["kubernetes.digitalocean.com/load-balancer-id"]
+	_, _, err := doClient.LoadBalancers.Update(ctx, currentLBID, lbRequest)
+	if err != nil {
+		fmt.Printf("THIS IS AN ERROR: %v", err)
```

Comment:
No need to log here since we do it at the call site already.

The whole block can be simplified to 

```go
_, _, err := doClient.LoadBalancers.Update(ctx, currentLBID, lbRequest)
return err
```

Similar for `validateCreate`.